projects/canopy/src/lib/carousel/carousel.component.ts

```diff
@@ -91,16 +100,18 @@ export class LgCarouselComponent implements AfterContentInit, OnDestroy {
   }
 
   ngAfterContentInit(): void {
+    this.selectedItemIndexSet$.subscribe((itemIndex) => {
+      this.selectedItemIndex = itemIndex;
+    });
     this.carouselItemCount = this.carouselItems.length;
     this.selectCarouselItem(0);
     if (this.autoPlay) {
       this.setAutoPlayInterval();
     }
   }
 
-  ngOnDestroy(): void {
-    this.unsubscribe.next();
-    this.unsubscribe.complete();
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
```

Comment:
Is.unsubscribe is used in the setAutoplayInterval on line 92.` takeUntil(this.unsubscribe)` Removing this will cause a potential memory leak when the component is destroyed.  I suggest you bring it back and use `takeUntil(this.unsubscribe)` as part of the `this.selectedItemIndexSet$` subscription on line 103